cmd/promtool/backfill.go

```diff
@@ -0,0 +1,193 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"math"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/textparse"
+	"github.com/prometheus/prometheus/tsdb"
+	tsdb_errors "github.com/prometheus/prometheus/tsdb/errors"
+)
+
+// OpenMetricsParser implements textparse.Parser. Use NewOpenMetricsParser to create new instances.
+type OpenMetricsParser struct {
+	textparse.Parser
+	s *bufio.Scanner
+}
+
+// NewOpenMetricsParser returns an OpenMetricsParser reading from the provided reader.
+func NewOpenMetricsParser(r io.Reader) *OpenMetricsParser {
+	return &OpenMetricsParser{s: bufio.NewScanner(r)}
+}
+
+// Next advances the parser to the next sample. It returns io.EOF if no
+// more samples were read.
+func (p *OpenMetricsParser) Next() (textparse.Entry, error) {
+	for p.s.Scan() {
+		line := p.s.Bytes()
+		line = append(line, '\n')
+
+		p.Parser = textparse.New(line, string(expfmt.FmtOpenMetrics))
+		if et, err := p.Parser.Next(); err != io.EOF {
+			return et, err
+		}
+	}
+
+	if err := p.s.Err(); err != nil {
+		return 0, err
+	}
+	return 0, io.EOF
+}
+
+func getMinAndMaxTimestamps(p textparse.Parser) (int64, int64, error) {
+	var maxt, mint int64 = math.MinInt64, math.MaxInt64
+
+	for {
+		entry, err := p.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return 0, 0, errors.Wrap(err, "next")
+		}
+
+		if entry != textparse.EntrySeries {
+			continue
+		}
+
+		_, ts, _ := p.Series()
+		if ts == nil {
+			return 0, 0, errors.Errorf("expected timestamp for series got none")
+		}
+
+		if *ts > maxt {
+			maxt = *ts
+		}
+		if *ts < mint {
+			mint = *ts
+		}
+	}
+
+	if maxt == math.MinInt64 {
+		maxt = 0
+	}
+	if mint == math.MaxInt64 {
+		mint = 0
+	}
+
+	return maxt, mint, nil
+}
+
+func createBlocks(input *os.File, mint, maxt, maxSamplesInAppender int64, outputDir string) error {
+
+	blockDuration := tsdb.DefaultBlockDuration
+	mint = blockDuration * (mint / blockDuration)
+
+	for t := mint; t <= maxt; t = t + blockDuration {
+
+		err := func() error {
+			// TODO(aSquare14) : Print block metadata.
+			w, err := tsdb.NewBlockWriter(log.NewNopLogger(), outputDir, blockDuration)
+			if err != nil {
+				return errors.Wrap(err, "block writer")
+			}
+			defer func() {
+				err = tsdb_errors.NewMulti(err, w.Close()).Err()
+			}()
+
+			ctx := context.Background()
+			app := w.Appender(ctx)
+
+			if _, errReset := input.Seek(0, 0); errReset != nil {
```

Comment:
```suggestion
			if _, err := input.Seek(0, 0); err!= nil {
```

No need to add complexity of another name, even more with name totally unrelated to the method.

Reply:
agreed. 👍